db/StoreArchive.py

```diff
@@ -0,0 +1,55 @@
+import FetchArchive as fetch
+
+from datetime import datetime
+from pymongo import MongoClient
+import os
+import logging
+
+if not os.path.exists('logs'):
+    os.makedirs('logs')
+logging.basicConfig(filename='logs/StoreArchive.log', level=logging.INFO, format='%(asctime)s:%(levelname)s:%(message)s')
+
+db = MongoClient('localhost', 27017).WallStreetDB
+
+
+def buildIndex():
+    '''Gets, cleans and stores data from archiveIndexGenerator().'''
+    form = '%Y-%m-%dT%H:%M:%SZ'
+    for item in fetch.archiveIndexGenerator():
+        # TODO: Add lastModified field.
+        item['_id'] = item['identifier']
+        key = {'_id': item['_id']}
+        if 'date' in item:  # Clean index date. (str -> datetime)
+            item['datetime'] = datetime.strptime(item['date'], form)
+        # Upsert inserts (instead of update) if the item does not exist.
+        db.ArchiveIndex.update_one(key, {'$set': item}, upsert=True)
```

Comment:
TODO: Add last modified field.
Currently overwrites any data that already exists in the database. Instead we should detect whether the data already exists and stop web requests to archive (via Archive Index Generator) once we've added all NEW shows to our existing index.
We don't need to implement the detection feature in this PR, but a last_modified field can help diagnose issues or selectively update later.

Reply:
Marking Resolved. This probably doesn't really matter.